core/network/space.go

```diff
@@ -80,6 +80,37 @@ func (s SpaceInfos) AllSubnetInfos() (SubnetInfos, error) {
 	return subs, nil
 }
 
+// FanOverlaysFor returns any subnets in this network topology that are
+// fan overlays for the input subnet IDs.
+func (s SpaceInfos) FanOverlaysFor(subnetIDs IDSet) (SubnetInfos, error) {
+	if len(subnetIDs) == 0 {
+		return nil, nil
+	}
+
+	var located int
+	allOverlays := make(SubnetInfos, 0)
+
+	for _, space := range s {
+		for _, sub := range space.Subnets {
+			if subnetIDs.Contains(sub.ID) {
+				overlays, err := space.Subnets.GetByUnderlayCIDR(sub.CIDR)
+				if err != nil {
+					return nil, errors.Trace(err)
+				}
+				allOverlays = append(allOverlays, overlays...)
+
+				// If we have tested all of the inputs, we can exit early.
+				located++
+				if located >= len(subnetIDs) {
```

Comment:
Curious as to why the condition is `>=` instead of `==`? The `>` seems redundant as we only seem to be incrementing `located` in the method body.

Reply:
Just going for the iron-clad.